src/Microsoft.AspNetCore.Server.Kestrel/Internal/Http/SocketInput.cs

```diff
@@ -321,5 +341,73 @@ private static void ReturnBlocks(MemoryPoolBlock block, MemoryPoolBlock end)
                 returnBlock.Pool.Return(returnBlock);
             }
         }
+
+        private class BufferLengthConnectionController
```

Comment:
This design was also problematic when using a connection filter, since this involves two instances of `SocketInput`.  Refactoring will be pushed shortly.